app/services/mapping/goodsshipment/ConsignmentBuilder.scala

```diff
@@ -122,4 +150,15 @@ object ConsignmentBuilder {
     )
     codeType
   }
+
+  private def isDefined(carrierDetails: CarrierDetails, cacheMap: CacheMap): Boolean =
+    carrierDetails.details.eori.getOrElse("").nonEmpty ||
+      (carrierDetails.details.address.isDefined && carrierDetails.details.address.get.isDefined()) ||
+      (cacheMap
+        .getEntry[TransportInformation](TransportInformation.id)
+        .isDefined && isTransportInformationDefined(
+        cacheMap
```

Comment:
You can only once retrieve this data from cache, save it in variable and reuse in both places. You don't need to getEntry two times

Reply:
I do not see this as a problem. because `cacheMap` is just a map in memory.
We are losing some efficiency in transversing through a `Map` structure instead of just referring to the reference, but I prefer to have that performance hit and have a more readable and stable code pattern.
If you think this is something that we should be doing, then I will address that on my next commit/PR.

Please let me know ;)